Planner/app/src/main/java/rdatu/android/cyscorpions/com/projectplanner/view/ScheduleTaskActivity.java

```diff
@@ -0,0 +1,278 @@
+                    } else {
+                        for (int i = timeStart; i < timeEnd; i++) {
+                            saveTasks(date, name, descr, place, priority, getFormattedTime(i));
+                        }
+                    }
+
+                } else {
+                    Toast.makeText(getApplicationContext(), "All fields are required.", Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+    }
+
+
+    private void saveTasks(String date, String name, String descr, String place, String priority, String time) {
+        Tasks task = new Tasks();
+        task.setDate(date);
+        task.setTaskName(name);
+        task.setDescription(descr);
+        task.setPlace(place);
+        task.setPriority(priority);
+        task.setTimeSlot(time);
+        mTaskManager.saveTask(task);
+        finish();
+
+    }
+
+    private void showDatePicker() {
+        FragmentManager fm = getSupportFragmentManager();
+        DatePickerDialog dialog = DatePickerDialog.newInstance(mDateButton.getText().toString(), FUNCTION_FORCHANGE);
+        dialog.show(fm, "datePicker");
+    }
+
+    private void showTimePicker() {
+        FragmentManager fm = getSupportFragmentManager();
+        TimePickerDialog dialog = TimePickerDialog.newInstance(mToTimeText.getText().toString());
+        dialog.show(fm, "timePicker");
+    }
+
+    private void showOverwriteDialog(String name, String descr, String place, String priority, String date, int start, int end) {
+        FragmentManager fm = getSupportFragmentManager();
+        OverwriteDialog dialog = OverwriteDialog.newInstance(name, descr, place, priority, date, start, end);
+        dialog.show(fm, "overWrite");
+    }
+
+    @Override
+    public void onTimeChanged(String time) {
+        mToTimeText.setText(time);
+        mTimeEnd = mToTimeText.getText().toString();
+    }
+
+    @Override
+    public void onDateChanged(String date) {
+        mDateButton.setText(date);
+    }
+
+    @Override
+    public void onJumpTo(String date) {
+        //Not used in this Activity
+    }
+
+    private boolean areFieldsFilled() {
+        boolean ready = true;
+
+        ready = isNotEmpty(mTaskNameText);
+        ready = isNotEmpty(mTaskDescriptionText);
+        ready = isNotEmpty(mPlaceText);
+
+        if (TextUtils.isEmpty(mFromTimeText.getText().toString()))
+            ready = false;
+        if (TextUtils.isEmpty(mToTimeText.getText().toString()))
+            ready = false;
+        if (TextUtils.isEmpty(mDateButton.getText().toString()))
+            ready = false;
+        if (TextUtils.isEmpty(mPriorityButton.getText().toString()))
+            ready = false;
+
+        return ready;
+    }
+
+    private boolean isNotEmpty(EditText item) {
+        if (TextUtils.isEmpty(item.getText().toString())) {
+            item.setError("This item must not be Empty");
+            return false;
+        }
+        return true;
```

Comment:
instead of having a conditional statement here, we can refactor it to something like this:

```return !TextUtils.isEmpty```

though since we have additional processs. We can assign the value to a variable, perform the process and return the value